src/lib/formatter/RuleResultRecombinator.ts

```diff
@@ -11,28 +12,27 @@ import { stringify } from 'csv-stringify';
 import { constructSarif } from './SarifFormatter';
 import {isPathlessViolation} from '../util/Utils';
 
-type BaseTableRow = {
-	Rule: string;
-	Description: string;
-	URL: string;
-	Category: string;
-	Severity: number;
-	Engine: string;
+const BASE_COLUMNS: Ux.Table.Columns<ResultTableRow> = {
+	description: {},
+	category: {},
+	url: {
+		header: "URL"
+	}
 };
 
-type PathlessTableRow = BaseTableRow & {
-	Location: string;
-	Line: number;
-	Column: number;
+const DFA_COLUMNS: Ux.Table.Columns<ResultTableRow> = {
+	sourceLocation: {
+		header: 'Source Location'
+	},
+	sinkLocation: {
+		header: 'Sink Location'
+	},
+	...BASE_COLUMNS
 };
 
-type DfaTableRow = BaseTableRow & {
-	"Source Location": string;
-	"Source Line": number;
-	"Source Column": number;
-	"Sink Location": string|null;
-	"Sink Line": number|null;
-	"Sink Column": number|null;
+export const PATHLESS_COLUMNS: Ux.Table.Columns<ResultTableRow> = {
+	location: {},
+	...BASE_COLUMNS
```

Comment:
The old row types had a bunch of properties defined that weren't actually being displayed, so I removed those.